crates/sui-config/src/node.rs

```diff
@@ -543,6 +543,13 @@ pub struct AuthorityStorePruningConfig {
     /// number of epochs to keep the latest version of transactions and effects for
     #[serde(skip_serializing_if = "Option::is_none")]
     pub num_epochs_to_retain_for_checkpoints: Option<u64>,
+    /// enables pruner to prune no longer needed object tombstones.
+    #[serde(default = "default_enable_pruning_tombstones")]
```

Comment:
Hi @phoenix-o @mystenmark , is this the right way to make this config optional as an input, and `always` have a default? (In this way, in the code where this config is accessed, it doesn't need to consider the case where it is not set).

I tried this locally and it doesn't crash anymore. I also removed some config setting I added in #14373  to make test passing, since this config now is no longer required in the config file.

Reply:
yes, it works, but you can also use the shorter version `#[serde(default)]` since you are just setting it to false. Also, you can use the same approach as for the `num_epochs_to_retain_for_checkpoints` parameter. This way, you won't have to change snapshot files

Reply:
Thanks for the pointer @phoenix-o ! Use `serde(default)` now and also skip serialization if the value is false (and no more changes in the snapshot file).